### PR TITLE
Migrate MavenV2 to Node20

### PR DESCRIPTION
**Pull request by qianz2**
**Task name**: MavenV2

**Description**: Migrate MavenV2 to Node20

**Documentation changes required:** (Y/N) <Please mark if documentation changes are required>

**Added unit tests:** (Y/N) <Please mark if unit tests were added or updated according changes>

**Attached related issue:** (Y/N) <Please add link to related issue here>

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [x] Checked that applied changes work as expected

### CODE REVIEW

File: make.js [hunk 1]
```diff
@@ -454,6 +449,21 @@ function buildTask(taskName, taskListLength, nodeVersion) {
             rm('-Rf', taskTestsNodeModulesPath);
         }
     }
+
+    // remove duplicated task libs node modules from build tasks.
+    var buildTasksNodeModules = path.join(buildTasksPath, taskName, 'node_modules');
```
Comment (DmitriiBobreshev) [inline make.js:454]: Shouldn't we remove duplicate task-lib via the [make.json file](https://github.com/microsoft/azure-pipelines-tasks/blob/master/Tasks/MavenV2/make.json#L5)?

File: make.js [hunk 2]
```diff
@@ -1021,8 +1031,22 @@ CLI.gentask = function() {
             console.log(`Running \"npm install\" command in ${taskPath}`);
             run(`npm install`);
             cd(__dirname);
+
+            // copy package.json, package-lock.json and npm-shrinkwrap.json files from generated tasks to the Tasks\taskname\_buildConfig\nodeversion folder.
+            const fileNames = ['package.json', 'package-lock.json', 'npm-shrinkwrap.json'];
```
Comment (DmitriiBobreshev) [inline make.js:1036]: maybe better to reuse [this function ](https://github.com/microsoft/azure-pipelines-tasks/blob/master/make-util.js#L1902) it's doing the same thing but with the build command(probably we need to add npm-shrinkwrap.json file to allowedFilesToCopy)?

File: make.js (hunk 1 above)
Comment (qianz2) [reply make.js:454]: Removing duplicated task-lib via make.json file would require us to add it under Tasks\taskname\_buildConfigs files for each task and identify all tasks that are having duplicated task-libs. I prefer to remove duplicates via buildTask function.

File: make.js (hunk 2 above)
Comment (qianz2) [reply make.js:1034]: Files are not getting copied after adding syncGeneratedFilesWrapper here, probably something different between logics here. I'll keep the logics here for consistency and get back to this.

File: make.js [hunk 3]
```diff
@@ -185,8 +185,6 @@ CLI.build = function(/** @type {{ task: string }} */ argv)
     if (process.env.TF_BUILD) {
         fail('Please use serverBuild for CI builds for proper validation');
     }
-
-    callGenTaskDuringBuild = true;
```
Comment (merlynomsft) [inline make.js:189]: please restore the generated tasks in _generated\* for the maven PR, we'll check in those changes seperately

**Review by merlynomsft — DISMISSED** [review]
:clock1:

File: make-options.json [hunk 4]
```diff
@@ -227,7 +227,9 @@
         "NuGetAuthenticateV0",
         "AzureFileCopyV2",
         "AzureFileCopyV3",
-        "DockerComposeV0",
-        "AzureRmWebAppDeploymentV3"
```
Comment (merlynomsft) [inline make-options.json:231]: AzureRmWebAppDeploymentV3 removed?  merge issue?

Comment (qianz2) [reply make-options.json:231]: Updated

File: make.js (hunk 3 above)
Comment (qianz2) [reply make.js]: Updated!

**Review by merlynomsft — APPROVED** [review]
:shipit:

**Review by tkasparek — APPROVED** [review]

**Pull request merged**